clouddriver-eureka/src/main/groovy/com/netflix/spinnaker/clouddriver/eureka/deploy/ops/AbstractEurekaSupport.groovy

```diff
@@ -139,18 +142,26 @@ abstract class AbstractEurekaSupport {
 
           Response resp
 
-          if (discoveryStatus == DiscoveryStatus.Disable) {
+          if (discoveryStatus == DiscoveryStatus.OUT_OF_SERVICE) {
             resp = eureka.updateInstanceStatus(applicationName, instanceId, discoveryStatus.value)
           } else {
-            resp = eureka.resetInstanceStatus(applicationName, instanceId, DiscoveryStatus.Disable.value)
+            // If we're trying to set the status to UP, and the instance is already UP (or hasn't registered yet),
+            // skip it.
+            if (instanceDetails.instance.status != DiscoveryStatus.OUT_OF_SERVICE.value) {
+              log.debug("Instance ${instanceId} is already UP in discovery, skipping.")
+              skipped.add(instanceId)
+              return
+            } else {
+              resp = eureka.resetInstanceStatus(applicationName, instanceId, DiscoveryStatus.OUT_OF_SERVICE.value)
+            }
```

Comment:
Minor: I had to read this a couple times to parse the double negative ("if not disabled then skip, else do the thing") so my preference would be to flip the branches.

Also changed the log message a little to provide the actual status.

```suggestion
            if (instanceDetails.instance.status == DiscoveryStatus.OUT_OF_SERVICE.value) {
              resp = eureka.resetInstanceStatus(applicationName, instanceId, DiscoveryStatus.OUT_OF_SERVICE.value)
            } else {
              log.debug("Instance {} is {} in discovery, no override to remove", instanceId, instanceDetails.instance.status)
              skipped.add(instanceId)
              return
            }
```

Reply:
Makes sense. BTW, why do we need to reset the status to `OUT_OF_SERVICE` when the instance is already marked `OUT_OF_SERVICE`? What exactly does that achieve?

Reply:
resetInstanceStatus *deletes* the `OUT_OF_SERVICE` override